File: src/main/java/com/wanted/preonboarding/layered/domain/entity/BaseEntity.java
package com.wanted.preonboarding.layered.domain.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Converter;
import jakarta.persistence.MappedSuperclass;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.SQLRestriction;

import java.time.LocalDateTime;

@Setter
@MappedSuperclass
public class BaseEntity {
    @Column(nullable = false, name = "created_at")
    private LocalDateTime createdAt;
    @Column(nullable = false, name = "updated_at")
    private LocalDateTime updatedAt;
    @Column(nullable = true, name = "deleted_at")
    private LocalDateTime deletedAt;
}

File: src/main/kotlin/com/wanted/preonboarding/ticket/infra/entity/BaseEntity.kt
package com.wanted.preonboarding.ticket.infra.entity

import jakarta.persistence.Column
import jakarta.persistence.MappedSuperclass
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime

@MappedSuperclass
abstract class BaseEntity {
    @CreationTimestamp
    @Column(updatable = false)
    val createdAt: LocalDateTime? = null

    @UpdateTimestamp
    @Column
    var updatedAt: LocalDateTime? = null
        protected set
}
